### DISCUSSION TITLE
conditionQualifiedbyName not working while conversion from MAP to Entity

### DISCUSSION THREAD

**Discussion by Manish210194** [mapstruct/mapstruct#3517] [Q&A]
conditionQualifiedbyName not working while conversion from MAP to Entity

1. how can we add condition when converting a map into entity?
2. is it possible to convert Map to Map using different Mapping 
 for ex:   @Mapping(source = "a",target = "b")
    @Mapping(source = "c",target = "d")
    HashMap<String,String> sourceToDestination(Map<String,Object> map);

**Comment by thunderhook** [comment]
Hi @Manish210194 

1. Here is a working example of a condition for the mapper adapted from [this test mapper](https://github.com/mapstruct/mapstruct/blob/main/processor/src/test/java/org/mapstruct/ap/test/frommap/MapToBeanDefinedMapper.java#L20-L23):

```java
@Mapping(target = "normalInt", source = "number", conditionQualifiedByName = "testCondition")
Target toTarget(Map<String, Integer> source);

@Named("testCondition")
@Condition
default boolean testCondition(Map<String, Integer> source) { // this has to be the same type as the parameter from the mapping method
    return source.size() % 2 == 0;
}
```

Please tell me if that worked for you.

2. This is not yet supported but has already been requested with the following issue:
* #3303
There is already an open merge request but more time is needed, therefore we won't support this in the 1.6.0 release. I expect it with version 1.7.0. For more details you can have a look at the merge request:
* #3367

**Reply by Manish210194** [reply]
Hi @thunderhook ,

Thanks for the quick reply, 
 1. how can we add condition when converting a map into entity?(Working now ), I had 2 parameters in the method, 

2. Could you please provide the tentative release date for version 1.7.0? We encountered a use case that required HashMap<String,String> sourceToDestination(Map<String,Object> map) conversion. 

3. One last question: is it feasible to map multiple sources to one target when mapping from MAP?
for ex: 
@Mapping(target = "B", source = "A")
@Mapping(target = "B", source = "C")
Target toTarget(Map<String, Integer> source);

merge source A and C and set it into target B based on condition


4. Is it possible to provide additionalProperties support, similar to JSON, where extra data is placed into an additional field  in Mapstruct? 

Regards,
Manish

**Reply by filiphr** [reply]
>1. how can we add condition when converting a map into entity?(Working now ), I had 2 parameters in the method,

This was already answered by @thunderhook in https://github.com/mapstruct/mapstruct/discussions/3517#discussioncomment-8290865

>2. Could you please provide the tentative release date for version 1.7.0? We encountered a use case that required HashMap<String,String> sourceToDestination(Map<String,Object> map) conversion.

No we cannot provide a release date, because we don't know when we are going to do this. If this is something that you need, then you are free to work on a contribution that would add that and we can include that in an earlier version.

>3. One last question: is it feasible to map multiple sources to one target when mapping from MAP?

No this is not feasible. MapStruct only supports mapping to one target.

>4. Is it possible to provide additionalProperties support, similar to JSON, where extra data is placed into an additional field in Mapstruct?

Can you be a bit more specific here? Are you talking about `@JsonAnySetter` from Jackson? I am not sure how this would work with MapStruct, because we are not generating code tracking what has been mapped. I would suggest that you write this code by hand and share it how it would look like, before we can consider adding it to MapStruct.
